sys/net/gnrc/network_layer/ipv6/nib/_nib-slaac.c

```diff
@@ -162,10 +162,9 @@ void _remove_tentative_addr(gnrc_netif_t *netif, const ipv6_addr_t *addr)
         /* Cannot use target address as personal address and can
          * not change hardware address to retry SLAAC => use purely
          * DHCPv6 instead */
-        /* TODO: implement IA_NA for DHCPv6 */
-        /* then => tgt_netif->aac_mode = GNRC_NETIF_AAC_DHCP; */
-        DEBUG("nib: would set interface %i to DHCPv6, "
-              "but is not implemented yet", netif->pid);
+
+        /* TODO: How will this flag be used? */
+        netif->ipv6.aac_mode = GNRC_NETIF_AAC_DHCP;
```

Comment:
> The comment says tgt_netif, so that should be used.

Oh, you are right, sorry! I will fix this immediately :)

> only interfaces with the DHCP flag should use DHCP IA_NA to fetch their address

I see, thanks for the clarification :) I was just wondering if the `aac_mode` field should be used or checked for somewhere else or if this more relevant for future implementations

Reply:
> I see, thanks for the clarification :) I was just wondering if the `aac_mode` field should be used or checked for somewhere else or if this more relevant for future implementations

Since this PR introduces IA_NA, it might be good if this is checked. But then again, from what I can see I did not seem to have thought of the case that one would use SLAAC _and_ DHCPv6 for address configuration (`aac_mode` is just an int, not a flag field), so maybe that field needs to be redone as a flag field.

Reply:
> The comment says `tgt_netif`, so that should be used.

Hmm, the signature of the function is

```c
void _remove_tentative_addr(gnrc_netif_t *netif, const ipv6_addr_t *addr)
```

should the `netif` be renamed there? So far no `tgt_netif` is present in the file which is why didn't use it.

> Since this PR introduces IA_NA, it might be good if this is checked. But then again, from what I can see I did not seem to have thought of the case that one would use SLAAC and DHCPv6 for address configuration (aac_mode is just an int, not a flag field), so maybe that field needs to be redone as a flag field.

Oh, good point. So in the end the line should look something like this

```c
    netif->ipv6.aac_mode |= GNRC_NETIF_AAC_DHCP;
```

I guess?

Reply:
> Hmm, the signature of the function is
> 
> ```c
> void _remove_tentative_addr(gnrc_netif_t *netif, const ipv6_addr_t *addr)
> ```
> 
> should the `netif` be renamed there? So far no `tgt_netif` is present in the file which is why didn't use it.

Ah, the function is called with `tgt_netif`, so its fine as is. Maybe the comment is from a time where this function was not a function.



> Oh, good point. So in the end the line should look something like this
> 
> ```c
>     netif->ipv6.aac_mode |= GNRC_NETIF_AAC_DHCP;
> ```
> 
> I guess?

Yepp, but also `GNRC_NETIF_AAC_DHCP` and `GNRC_NETIF_AAC_DHCP` should be specifically defined as flags. Currently, they are 1 and 2 respectively more or less by chance, but that could change.

Reply:
I have tried to add this via 3e91f680728cf67a673f559ddcd25667ef35b2fd now.